lib/elixir/lib/kernel/cli.ex

```diff
@@ -14,14 +22,17 @@ defmodule Kernel.CLI do
     {config, argv} = parse_argv(argv)
     System.argv(argv)
 
-    run fn _ ->
-      errors = process_commands(config)
+    run(
+      fn _ ->
+        errors = process_commands(config)
 
-      if errors != [] do
-        Enum.each(errors, &IO.puts(:stderr, &1))
-        System.halt(1)
-      end
-    end, config.halt
+        if errors != [] do
+          Enum.each(errors, &IO.puts(:stderr, &1))
+          System.halt(1)
+        end
+      end,
+      config.halt
+    )
```

Comment:
Please extract the function to a separate variable and call `run(x, config.halt)`.